azurelinuxagent/ga/utils/exthandler_utils.py

```diff
@@ -0,0 +1,343 @@
+        return self._data.get("version", None)
+
+    def get_linux_configurations(self):
+        return self._data['handlerConfiguration']['linux']
+
+    def get_resource_configurations(self):
+        return self.resource_config
+
+    def get_resource_limits(self):
+        return self.resource_limits
+
+
+class ExtensionResourcesConfiguration(object):
+    def __init__(self, resource_configuration):
+        self._data = resource_configuration["resources"]
+        self.cpu_limits = None
+        self.memory_limits = None
+
+        try:
+            if "cpu" not in self._data and "memory" not in self._data:
+                raise ExtensionHandlerConfigurationError('No "cpu" node present in '
+                                                  'ResourceConfiguration')
+
+            if "cpu" in self._data:
+                self.cpu_limits = CpuLimits(self._data["cpu"])
+            if "memory" in self._data:
+                self.memory_limits = MemoryLimits(self._data["memory"])
+        except ExtensionHandlerConfigurationError as e:
+            logger.warn(ustr(e))
+            HandlerConfiguration.send_handler_configuration_event(message=ustr(e), is_success=False,
+                                                                  log_event=False,
+                                                                  operation=WALAEventOperation.HandlerConfiguration)
+            raise e
+
+    def get_cpu_limits_for_extension(self):
+        return self.cpu_limits
+
+    def get_memory_limits_for_extension(self):
+        return self.memory_limits
+
+
+class CpuLimits(object):
+    def __init__(self, cpu_node):
+        self.cpu_limits = []
+        self.cores = []
+
+        for cpu_info in cpu_node:
+            # integers and > 0.
+            if "cores" not in cpu_info or "limit_percentage" not in cpu_info:
+                raise ExtensionHandlerConfigurationError("Malformed CPU limit node in HandlerConfiguration")
+            self.cpu_limits.append(CpuLimitInstance(cpu_info["cores"], cpu_info["limit_percentage"]))
+            self.cores.append(cpu_info["cores"])
+
+        if DEFAULT_CORES_COUNT not in self.cores:
+            raise ExtensionHandlerConfigurationError("Default CPU limit not set."
+                                              " Core configuration for {0} not present".format(DEFAULT_CORES_COUNT))
+
+        self.cpu_limits = sorted(self.cpu_limits)
+
+    def __str__(self):
+        return self.cpu_limits.__str__()
+
+
+class CpuLimitInstance(object):
+    def __init__(self, cores, limit_percentage):
+        if type(cores) is not int:
+            raise ExtensionHandlerConfigurationError(
+                "Incorrect types for CPU values | field - {0}/ value - {1}/ type - {2}".format("cores", cores,
+                                                                                               type(cores)))
+
+        if type(limit_percentage) is not float and type(limit_percentage) is not int:
```

Comment:
do you want to check strictly against float? (e.g. force "10.0" instead of "10")

Reply:
In this case, I felt it was okay to give `float` as well as `int` as a percentage value. I am just making sure it is a number that they pass. I don't want to be too strict with floats. People will forget the `.0` suffix.

Reply:
yup; agree - you should then try to parse the value (or something like that) instead of strict type check

>>> type(1)
<class 'int'>
>>> type(1.0)
<class 'float'>
>>>

Reply:
nevermind, i see your point now